packages/chat-headless/tests/chatheadless.test.ts

```diff
@@ -343,3 +346,58 @@ it("restartConversation works as expected", () => {
   };
   expect(chatHeadless.state).toEqual(expectedState);
 });
+
+describe("loadSessionState works as expected", () => {
+  const expectedState = {
+    conversationId: "dummy-id",
+    messages: [
+      {
+        text: "How can I help you?",
+        source: MessageSource.BOT,
+        timestamp: "2023-05-15T17:39:58.019Z",
+      },
+    ],
+    notes: {
+      currentGoal: "GOAL",
+    },
+    isLoading: true,
+  };
+  it("loads valid state from session storage", () => {
+    sessionStorage.setItem(
+      STATE_SESSION_STORAGE_KEY,
+      JSON.stringify(expectedState)
+    );
+    const chatHeadless = new ChatHeadless(config);
+    chatHeadless.setState({
+      ...chatHeadless.state,
+      meta: mockedMetaState,
+    });
+
+    expect(chatHeadless.state).toEqual({ conversation: expectedState, meta: mockedMetaState });
+  });
+
+  it("does not persist or load state when toggle is off", () => {
+    sessionStorage.setItem(
+      STATE_SESSION_STORAGE_KEY,
+      JSON.stringify(expectedState)
+    );
+    const chatHeadless = new ChatHeadless(config, false);
+    chatHeadless.setState({
+      ...chatHeadless.state,
+      meta: mockedMetaState,
+    });
+    expect(chatHeadless.state).toEqual({ conversation: initialState, meta: mockedMetaState });
+    const modifiedMessages = [
+      ...expectedState.messages,
+      {
+        text: "This is a new message",
+        source: MessageSource.USER,
+        timestamp: "2023-05-15T17:39:58.019Z",
+      },
+    ];
+    chatHeadless.setMessages(modifiedMessages);
+    expect(sessionStorage.getItem(STATE_SESSION_STORAGE_KEY)).toEqual(
```

Comment:
"does not persist or load state when toggle is off" don't we expect this in sessionStorage to be empty then? instead of equal to expectedState

Reply:
I set the sessionStorage state explicitly at the start of this test, then make a change to the ChatHeadless, and assert that the change did not propagate to the state in sessionStorage. But that's a bit confusing, so I will just leave it unset and assert that it's empty

Reply:
oh I see, eh, that's fine then